src/ImageSharp/Metadata/Profiles/Exif/ExifReader.cs

```diff
@@ -349,6 +394,95 @@ private void ReadValue(List<IExifValue> values)
+
+            ulong numberOfComponents = this.ReadUInt64();
+
+            this.TryReadSpan(this.offsetBuffer8);
+
+            if (dataType == ExifDataType.Unknown)
+            {
+                return;
+            }
+
+            if (dataType == ExifDataType.Undefined && numberOfComponents == 0)
+            {
+                numberOfComponents = 8;
+            }
+
+            // The StripOffsets, StripByteCounts, TileOffsets, and TileByteCounts tags are allowed to have the datatype TIFF_LONG8 in BigTIFF.
+            // Old datatypes TIFF_LONG, and TIFF_SHORT where allowed in the TIFF 6.0 specification, are still valid in BigTIFF, too.
+            // Likewise, tags that point to other IFDs, like e.g. the SubIFDs tag, are now allowed to have the datatype TIFF_IFD8 in BigTIFF.
+            // Again, the old datatypes TIFF_IFD, and the hardly recommendable TIFF_LONG, are still valid, too.
+            // https://www.awaresystems.be/imaging/tiff/bigtiff.html
+            ExifValue exifValue = null;
+            switch (tag)
+            {
+                case ExifTagValue.StripOffsets:
+                    exifValue = new ExifLong8Array(ExifTagValue.StripOffsets);
+                    break;
+                case ExifTagValue.StripByteCounts:
+                    exifValue = new ExifLong8Array(ExifTagValue.StripByteCounts);
+                    break;
+                case ExifTagValue.TileOffsets:
+                    exifValue = new ExifLong8Array(ExifTagValue.TileOffsets);
+                    break;
+                case ExifTagValue.TileByteCounts:
+                    exifValue = new ExifLong8Array(ExifTagValue.TileByteCounts);
+                    break;
+                ////case ExifTagValue.SubIFDOffset:
+                ////    exifValue = new ExifLong8Array(ExifTagValue.SubIFDOffset);
+                ////    break;
+                ////case ExifTagValue.GPSIFDOffset:
+                ////    exifValue = new ExifLong8Array(ExifTagValue.GPSIFDOffset);
+                ////    break;
+                default:
+                    exifValue = exifValue = ExifValues.Create(tag) ?? ExifValues.Create(tag, dataType, numberOfComponents);
+                    break;
+            }
+
+            if (exifValue is null)
+            {
+                this.AddInvalidTag(new UnkownExifTag(tag));
+                return;
+            }
+
+            ulong size = numberOfComponents * ExifDataTypes.GetSize(dataType);
+            if (size > 8)
+            {
+                ulong newOffset = this.ConvertToUInt64(this.offsetBuffer8);
+                if (newOffset > ulong.MaxValue || (newOffset + size) > (ulong)this.data.Length)
+                {
+                    this.AddInvalidTag(new UnkownExifTag(tag));
+                    return;
+                }
+
+                this.RegisterExtLoader(newOffset, () =>
+                {
+                    byte[] dataBuffer = new byte[size];
```

Comment:
Can the allocation for dataBuffer be avoided?
As well as the closure for size?

For dataBuffer stack-alloc or use a buffer from a pool?
To avoid the closure `RegisterExtLoader` should have a generic state argument, which allows to pass in the state.

But the question is: This adds some more complexity, so is it worth?

Reply:
@gfoidl Thanks for your thoughtful review. This code confuses me too and definitely needs to be improved. BufferPool would be more suitable for this code, but so far it isn't used in Exif. Usually Exif contains a little data, so this is not critical. Closing will not happen and the state will not be damaged either, this is an internal class and the extLoaders are guaranteed called before the stream closing. But from the outside perspective the code really looks questionable, I will try to improve this code.